versioned_hdf5/wrappers.py

```diff
@@ -963,7 +967,7 @@ def size(self):
     def fillvalue(self):
         if self._fillvalue is not None:
             return np.array([self._fillvalue], dtype=self.dtype)[0]
-        if self.dtype.metadata:
+        if getattr(self.dtype, "metadata", None):
```

Comment:
What's the difference between this and what was there previously? From what [I can tell from the docs](https://numpy.org/doc/stable/reference/generated/numpy.dtype.metadata.html) it seems like this is always defined on the dtype, although it can have the value of `None`.

Reply:
At some point I swear I got an AttributeError. Can't reproduce now. Reverting.

Reply:
Ah, there you go.
On numpy 1.x, `np.int16` doesn't have a `.metadata` attribute, but `np.asarray(0, dtype=np.int16).dtype` does.